File: typescript/src/airbnb.ts
import Computer from 'tzafon';
import path from 'node:path';
import { ensureDir, timestamp, downloadToFile } from './utils.js';

async function run() {
  const outDir = path.join(path.dirname(new URL(import.meta.url).pathname), '../results/airbnb');
  ensureDir(outDir);

  const client = new Computer();
  const computer = await client.create({ kind: 'browser' });
  await computer.navigate('https://www.airbnb.com/');
  try { await computer.wait(2); } catch {}
  const result = await computer.screenshot();
  const url = (result as any)?.result?.screenshot_url as string | undefined;
  console.log(`Screenshot: ${url}`);
  if (url) {
    const img = path.join(outDir, `${timestamp('airbnb_')}.png`);
    await downloadToFile(url, img);
    console.log(`Saved: ${img}`);
  }
}

run();

File: python/airbnb.py
import os
from tzafon import Computer
from utils import ensure_dir, timestamp, download_to_file


def run() -> None:
    out_dir = os.path.join(os.path.dirname(__file__), "results", "airbnb")
    ensure_dir(out_dir)

    client = Computer()
    computer = client.create(kind="browser")
    computer.navigate("https://www.airbnb.com/")
    try:
        computer.wait(2)
    except Exception:
        pass

    result = computer.screenshot()
    try:
        url = result.result["screenshot_url"]
    except Exception:
        url = None
    print(f"Screenshot: {url}")
    if url:
        img = os.path.join(out_dir, f"{timestamp('airbnb_')}.png")
        download_to_file(url, img)
        print(f"Saved: {img}")


if __name__ == "__main__":
    run()
